### DISCUSSION TITLE
Feature: Restricted Molecular Dynamics Simulation

### DISCUSSION THREAD

**Discussion by WHUweiqingzhou** [deepmodeling/abacus-develop#5262] [General]
### Background

Can ABACUS perform Molecular Dynamics Simulation under certain restrictions?

### Describe the solution you'd like

please support it.

### Task list only for developers

- [ ] Notice possible changes of behavior
- [ ] Explain the changes of codes in core modules of ESolver, HSolver, ElecState, Hamilt, Operator or Psi

### Task list for Issue attackers (only for developers)

- [ ] Review and understand the proposed feature and its importance.
- [ ] Research on the existing solutions and relevant research articles/resources.
- [ ] Discuss with the team to evaluate the feasibility of implementing the feature.
- [ ] Create a design document outlining the proposed solution and implementation details.
- [ ] Get feedback from the team on the design document.
- [ ] Develop the feature following the agreed design.
- [ ] Write unit tests and integration tests for the feature.
- [ ] Update the documentation to include the new feature.
- [ ] Perform code review and address any issues.
- [ ] Merge the feature into the main branch.
- [ ] Monitor for any issues or bugs reported by users after the feature is released.
- [ ] Address any issues or bugs reported by users and continuously improve the feature.

**Comment by kirk0830** [comment]
The most common way to constraint one atom's position is directly making the force to be zero, I think this is very easy to impl., but not sure if there will be any risks...

**Comment by QuantumMisaka** [comment]
@kirk0830 is PMF one of the most common way for C-MD  ?

**Comment by kirk0830** [comment]
> @kirk0830 is PMF one of the most common way for C-MD  ?

@QuantumMisaka not sure about the context, even a common MD with some atoms fixed can be called a constraint MD.
